test_collect-logs.py

```diff
@@ -360,6 +365,27 @@ def test_success(self):
                                       stderr=subprocess.STDOUT,
                                       env=None,
                                       ))
+        # for _create_ps_mem_output_file()
+        for host in self.hosts:
```

Comment:
I've refactored the test to better exercise the new methods under a separate class. The test code to look for a long list of expected `subprocess.check_output` calls has been removed.